### PR TITLE
Make Node URL available.

### PR DESCRIPTION
**Pull request by platfowner**
Summary:
- Make Node URL available from transaction pool and functions
- Fix undefined 'action' error

#107

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -173,10 +173,15 @@ class P2pServer {
     })
     .then((ipAddr) => {
       this.ipAddress = ipAddr;
+      this.node.setNodeUrl(this.getNodeUrl(ipAddr))
```
Comment (liayoo) [inline server/index.js:176]: Actually, when I tested, I needed to use an internal IP address for HOSTING_ENV === 'gcp'. Here's an URL we can get the IP from: http://metadata.google.internal/computeMetadata/v1/instance/network-interfaces/0/ip

**Review by liayoo — COMMENTED** [review]
Thanks for the changes! Left a comment regarding the gcp env, PTAL.

File: server/index.js (hunk 1 above)
Comment (platfowner) [reply server/index.js]: Thanks, just made both internal and external addresses are available.

**Comment by platfowner** [comment]
PTAL @liayoo

**Review by liayoo — APPROVED** [review]
LGTM, thanks!

**Pull request merged**